scripts/cat/thoughts.py

```diff
@@ -94,7 +95,36 @@ def cats_fulfill_thought_constraints(main_cat, random_cat, thought, game_mode, b
                 return False
         elif 'random_status_constraint' in thought and not random_cat:
             pass
+        
+        # Constraints for empty positions of power in the Clan
+        # Only works for leader, deputy, and medicine cat
+        if 'clan_missing_status_constraint' in thought:
+            for status in thought['clan_missing_status_constraint']:
+                if status == 'medicine cat':
+                    missing_status = "med_cat_list"
+                else:
+                    missing_status = status
+                if game.clan:
+                    if getattr(game.clan, missing_status):
+                        print("FAIL", main_cat.name, missing_status, getattr(game.clan, missing_status))
+                        return False
+                    else:
+                        print("PASS", main_cat.name, missing_status, getattr(game.clan, missing_status))
 
+        # Constraints for size of herb store
+        if 'herb_max_constraint' in thought:
+            if game.clan:
+                herb_max = (thought['herb_max_constraint'])[0]
+                if herb_max < sum(game.clan.herbs.values()):
```

Comment:
Regarding calculating how many herbs a clan needs, I'd direct you toward the calculations used in ShortEvent filtering.  ShortEvents allow constraining to set supply amounts of both certain herbs and overall herb supply: low, adequate, full, excess, as well as an "always" to trigger no matter how many herbs are in the supply.  I'll admit the filtering is a bit of a headache to look at, so if you want some help understanding it then let me know.

This is admittedly something I'm still trying to balance sufficiently.  If we're going to have it determined in multiple places, then there should be consideration towards either a function or some game_config settings so that balancing can remain consistent and easy to alter in the future.

Reply:
I tried adding some utility functions based off of the values in the medicine cat concern string (where it shows what they think of their current stores), is that more like what you had in mind?

Reply:
Yeah, very nice!  Could you also allow for checking the supply of a singular herb?